### ISSUE TITLE
scenario : unicode error

### ISSUE THREAD

**Issue by fritz-smh** [domogik/domogik#363]
```
2016-11-02 23:16:39,293 domogik-scenario.Notification maison DEBUG Evaluate SensorTest '330' in mode 'value' to 'Fred n'est plus à la maison'. Type is '<type 'unicode'>'
2016-11-02 23:16:39,295 domogik-scenario.Notification maison ERROR Error while evaluating condition '<code object <module> at 0x7fc5a80ae2b0, file "Scenario Notification maison", line 1>'. Error is : Traceback (most recent call last):
  File "/opt/dmg/domogik/src/domogik/scenario/scenario.py", line 408, in eval_condition
    exec(self._compiled_condition)
  File "Scenario Notification maison", line 9, in <module>
UnicodeEncodeError: 'ascii' codec can't encode character u'\xe0' in position 16: ordinal not in range(128)

```

**Comment by Cereal2nd** [comment]
hmm, to solve this i need the json for the scenario

**Comment by fritz-smh** [comment]
Here it is :

```
2016-11-07 03:13:58,072 domogik-scenario.Notification maison DEBUG Scenario 'Notification maison' python generated code :
print('---- Start evaluating ----')
if self._mapping['test']['01aed13b-bccf-4a74-b554-7dce6aec7818'].evaluate():^M
    print('---- Start evaluating ----')
    if ( ( self._mapping['test']['61957848-2166-4bbc-85b1-edc4d7a3271d'].evaluate() > self._mapping['test']['8841e280-db7f-463a-b4b2-0ef9142edc2a'].evaluate() ) and ( self._mapping['test']['d2f52d5f-a95b-4277-bc84-bef347b69ebe'].evaluate() < self._mapping['test']['30d646d7-53a0-4616-8af6-6c219c7f5aac'].evaluate() ) ):^M
            self._mapping['action']['0-cf81cb64-1539-4075-8db1-41a26fcfbf5f'].set_param("message", (str("Notification maison : ") + str(self._mapping['test']['4b61b16f-512b-48ee-af28-6f6893d7f4bf'].evaluate())))^M
            self._mapping['action']['0-cf81cb64-1539-4075-8db1-41a26fcfbf5f'].do_action()^M
            self._mapping['action']['1-103ac7ab-2772-4f66-801c-97f05d16cd8c'].set_param("text", (self._mapping['test']['a4840bf4-585f-4cfe-b33b-9d3d0d7b9a50'].evaluate()))^M
            self._mapping['action']['1-103ac7ab-2772-4f66-801c-97f05d16cd8c'].do_action()^M
    self._mapping['action']['2-d2ad8fba-fce2-43f1-8b4a-318e3eb87240'].set_param("url", (str("https://smsapi.free-mobile.fr/sendmsg?user=xxxxxxxx&pass=xxxxxxxxxxxxxxxmsg=") + str("Notif maison : ") + str(self._mapping['test']['046501cb-dab3-4aca-9813-6ee97b55f6f1'].evaluate())))^M
    self._mapping['action']['2-d2ad8fba-fce2-43f1-8b4a-318e3eb87240'].do_action()^M

```

**Comment by fritz-smh** [comment]
```
{u'IF0': {u'type': u'sensor.SensorTest.330', u'id': u'yaJDw1yUX^~A(l7Ggfzk', u'usage.usage': u'trigger_on_all'}, u'DO0': {u'IF0': {u'A': {u'A': {u'type': u'time_now.TimeNow', u'id': u'1jfQ6?2xCuyliAS9cpj^'}, u'B': {u'type': u'time_hhmm.TimeHHMMTest', u'id': u'k%~qp3Yo:HO_o.Fpt!e]', u'time.time': u'07:20'}, u'type': u'logic_compare', u'id': u'JfF9qb,,N73wNVgcW|H?', u'OP': u'GT'}, u'B': {u'A': {u'type': u'time_now.TimeNow', u'id': u',yikrSoc(y-p^KvDCBp?'}, u'B': {u'type': u'time_hhmm.TimeHHMMTest', u'id': u'Db-tPOnU):Kq.oHZYg-b', u'time.time': u'22:00'}, u'type': u'logic_compare', u'id': u'aobDJ4Xa_^Jqj9w3wPa[', u'OP': u'LT'}, u'type': u'logic_operation', u'id': u'egO!My9[3,Bdkl*qR`)%', u'OP': u'AND'}, u'DO0': {u'message': {u'itemCount': 2, u'type': u'text_join', u'id': u':?z;lMTeXsrLwn.NC9g?', u'ADD1': {u'type': u'sensor.SensorTest.330', u'id': u'2_D-rJQYRHHpZT#h:Xi{', u'usage.usage': u'value'}, u'ADD0': {u'TEXT': u'Notification maison : ', u'type': u'text', u'id': u'L?!rDn*]I;)QpD/J?E~;'}}, u'type': u'log.LogAction', u'id': u's/]..:#*DWVyQJPL!T6!', u'NEXT': {u'text': {u'type': u'sensor.SensorTest.330', u'id': u'kPv-LhogQwL]3[ScxY3J', u'usage.usage': u'value'}, u'type': u'command.CommandAction.37', u'id': u'R11nN}Vo?ZVL;!j@A?:?'}}, u'type': u'controls_if', u'id': u'?4O)P[iy.%*NK/NQ2HN_', u'NEXT': {u'url': {u'ADD2': {u'type': u'sensor.SensorTest.330', u'id': u'^ia-0uWKWGk]kSWw!a6o', u'usage.usage': u'value'}, u'ADD1': {u'TEXT': u'Notif maison : ', u'type': u'text', u'id': u'r%4=kzeD_?k{E3s=I6Hg'}, u'ADD0': {u'TEXT': u'https://smsapi.free-mobile.fr/sendmsg?user=xxxxxxxx&pass=xxxxxxxxxxxxxxxxxg=', u'type': u'text', u'id': u'y~l%z!zQeJnN4+`*l^!d'}, u'itemCount': 3, u'type': u'text_join', u'id': u'@GOR:N/4Plh!_9WolnvS'}, u'type': u'call_url.CallUrlAction', u'id': u'NLlBF!_?C)[g#_n=881;'}}, u'type': u'controls_if', u'id': u'CXeRaH*|O;%(XRY0~0(d'}

```

**Comment by Cereal2nd** [comment]
hmm, i don't see any special chars in the json ...

**Comment by Nico0084** [comment]
I had this kind of issue, but not save log :(
I presume special char could be in TEXT value of ADD0 : "user=xxxxxxxx&pass=xxxxxxxxxxxxxxxmsg".
You have probably modify for public data ;)

**Comment by tikismoke** [comment]
in the tracebiack position 16:
`Fred n'est plus à la maison`
The "à" no?

**Comment by Cereal2nd** [comment]
ok the problem is here ....

reslst.append(u"str({0})".format(addp))

by first converting it to a string we miss unicode chars.
To be honest i would like to handle this in 0.6, as this is a pretty big fix

**Comment by fritz-smh** [comment]
Not sure this is a big fix, but ok for 0.6
We need to release 0.5.

**Comment by fritz-smh** [comment]
Fixed in 0.6 with a previous commit... (no idea which one!)
